File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Indonesia.m
//
//  UIImage+Indonesia.m
//  CountriesFlags
//

#import "Countries/UIImage+Indonesia.h"

@implementation UIImage (Indonesia)

+ (UIImage *)flagIndonesiaWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor redColor];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0, size.height / 2.0, size.width, size.height / 2.0));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Indonesia.swift
//
//  UIImage+Indonesia.swift
//  CountriesFlags
//  

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Indonesia "wikipedia"
     
     Generates the flag of Indonesia.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Indonesia.
     */
    internal static func flagIndonesia(with size: CGSize) -> UIImage {
        let colorRed = UIColor.red
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 2.0),
                          size: CGSize(width: size.width, height: size.height / 2.0)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
